server/container_create.go

```diff
@@ -299,6 +300,26 @@ func generateUserString(username, imageUser string, uid *types.Int64Value) strin
 func (s *Server) CreateContainer(ctx context.Context, req *types.CreateContainerRequest) (res *types.CreateContainerResponse, retErr error) {
 	log.Infof(ctx, "Creating container: %s", translateLabelsToDescription(req.GetConfig().GetLabels()))
 
+	if req.SandboxConfig != nil && req.SandboxConfig.Metadata != nil && req.Config != nil && req.Config.Image != nil {
```

Comment:
AFAICS this does not work at all.

`req.Config.Image` is set, ultimately, from either CRI-O’s `ImageStatusResponse.Image.Id` (does not carry a repo information of any kind), or from CRI-O’s `PullImageResponse.ImageRef` (either an image ID again, or a `RepoDigests` value that might even point to a wrong repo).

Either way this does does not provide the information about the expected image identity, necessary to both select the right policy scope, and to correctly protect against image substitution.

At a first glance it seems to me that this would require a change in the CRI, to pass the original user-requested image name.